tsdb/index/index_test.go

```diff
@@ -242,18 +219,12 @@ func TestIndexRW_Postings(t *testing.T) {
 
 	// Test ShardedPostings() with and without series hash cache.
 	for _, cacheEnabled := range []bool{false, true} {
-		t.Run(fmt.Sprintf("ShardedPostings() cache enabled: %v", cacheEnabled), func(t *testing.T) {
-			var cache ReaderCacheProvider
-			if cacheEnabled {
-				cache = hashcache.NewSeriesHashCache(1024 * 1024 * 1024).GetBlockCacheProvider("test")
-			}
-
-			ir, err := NewFileReaderWithOptions(fn, cache)
-			require.NoError(t, err)
+		t.Run(fmt.Sprintf("ShardedPostings() cache enabled: %v", cacheEnabled), func(innerT *testing.T) {
```

Comment:
Why do you call it `innerT`? The standard idiom is to just call it `t`, thus shadowing the parent `t` (which is what you want, otherwise failures will be confusing). It's also a change from `main`, which I don't see why we would want.
```suggestion
		t.Run(fmt.Sprintf("ShardedPostings() cache enabled: %v", cacheEnabled), func(t *testing.T) {
```

Reply:
It was an attempt to fix the above - let me change it. I thought the shadowing of `t` was conflicting with the call to `t.Cleanup` but it turned out to not to be that problem.

Reply:
Done in https://github.com/grafana/mimir-prometheus/pull/639/commits/86722a6ef758dc731aa158c1792b9f4a10d3e828